sdk/src/store.rs

```diff
@@ -2332,38 +2332,51 @@ impl Store {
             }
         }
 
-        // we will not do automatic hashing if we detect a box hash present
         let mut needs_hashing = false;
-        if pc.hash_assertions().is_empty() {
-            // 2) Get hash ranges if needed, do not generate for update manifests
-            let mut hash_ranges = object_locations_from_stream(format, &mut intermediate_stream)?;
-            let hashes: Vec<DataHash> = if pc.update_manifest() {
-                Vec::new()
-            } else {
-                Store::generate_data_hashes_for_stream(
-                    &mut intermediate_stream,
-                    pc.alg(),
-                    &mut hash_ranges,
-                    false,
-                )?
-            };
+        // 2) If we have no hash assertions (and aren't an update manifest),
+        // add a hash assertion.  This creates a preliminary JUMBF store.
+        if pc.hash_assertions().is_empty() && !pc.update_manifest() {
+            needs_hashing = true;
+            if let Some(handler) = get_assetio_handler(format) {
+                // If our asset supports box hashing, generate and add a box
+                // hash assertion.
+                if let Some(box_hash_handler) = handler.asset_box_hash_ref() {
+                    let mut box_hash = BoxHash::new();
+                    box_hash.generate_box_hash_from_stream(
+                        &mut intermediate_stream,
+                        pc.alg(),
+                        box_hash_handler,
+                        false,
```

Comment:
If you set this parameter as true it will calculate the box hash.  Since the point of box hashes is that they can be pre-calculated you should not need the replace_box_hash function.  That mean you would not code below that does the fix up.

Reply:
So - we actually have some questions about the `minimal_form` functionality; if we're understanding the specification correctly, box hashes which have ranges _should_ include the data in between boxes, but otherwise data which is in between boxes shouldn't actually be included in the hash.  From the looks of the code, the `minimal_form` path actually concatenates all boxes, which would mean that data which was previously excluded (because it wasn't included in a range) would get calculated into the hash.  If this is the case, we think that the `minimal_form` behavior might be breaking the spec?

Do you know if this is not the right reasoning?

Reply:
Aside from that; even if we don't send `minimal_form` the `generate_box_hash_from_stream` call still does calculate the hashes for each box, but the preliminary hash may not be correct.   For instance, if the asset wasn't signed previously, the C2PA box may not exist (and adding it may modify offsets throughout the asset) - in these cases, I think you would still need to recalculate the hashes afterward?

This behavior is what caused us to remove the size comparison at the end `start_save` and `start_save_stream` because the preliminary store and actual store had different sizes (due to the addition of the C2PA box)

Reply:
We developed box hash to cover case where the caller wants to hash independent of where the c2pa box is but the order of the content is known.  For example a camera has hardware hasher that runs while they are saving the file.  They wanted to be able to specify the hashes during their writing of the info.  We expect the use of box hash to have added the box hash assertion.  Our SDK is designed for that use case.  We have not enabled writing of box hashes yet because of the issues you highlight and more.  

minimal_form was for the case when we generate box_hashes, which we don't at this time.  It looks like there is a bug in would affect writing (if it were enabled) PNG/JPEG in our code.  The get_box_map implementations are supposed to also insert a C2PA box into the map when not present to indicate where we would write the C2PA box.  It think if you returned a C2PA box from that function then what I described above should work since that is how we intended the box hash to work, i.e. user supplies where the manifest goes among the boxes.

Reply:
That makes sense - we had originally been returning a dummy C2PA box from `get_box_map` for our font implementation (as it made the JUMBF store check validate) but noticed that JPEG/PNG didn't have that behavior, and the comments didn't suggest it was expected.

So - we have a question on how we should move forward; we're actively trying to finish off the box hash assertion code in our font asset handler, and we believe this is the last issue we need to get sorted out.  So we're wondering:
  - Did you want us to modify the JPEG/PNG `get_box_map` calls to create dummy C2PA boxes as part of this PR?  Then we could put the JUMBF store check back in (theoretically) and get everything cleared out at once
  - Or.. does it make sense for us to modify the JPEG/PNG `get_box_map` code and stick it up in a separate PR?
  - Or.. do we need to contact someone else to get the JPEG/PNG code modifications, instead of us doing the modifications ourselves?

If it's easier, we can have a discussion over Discord to figure out what the best direction is, just let us know :)

I think down the road we should maybe talk more about the `minimal_form` behavior; but for now (as I think it's just an optimization) we can probably table that and just get the normal mode working as expected.

Reply:
Please file a bug for fixing JPEG/PNG.  We have to make sure we actually write out the C2PA box where we said we would place it.  Let's pick up the conversation on Discord on how to move forward.

Reply:
Sounds good; I'll write up the bug now and then contact you once I'm free from our morning meetings.

Reply:
Created https://github.com/contentauth/c2pa-rs/issues/429 - let me know if you need more detail or information on the issue and I'd be happy to add as much as needed

Reply:
Looks good.